tests/cpp_unit_tests/test_asym_line.cpp

```diff
@@ -0,0 +1,157 @@
+    double const loading = std::max(cabs(i1f), cabs(i1t)) / 200.0;
+
+    // Short circuit results
+    DoubleComplex const if_sc{1.0, 1.0};
+    DoubleComplex const it_sc{2.0, 2.0 * sqrt(3)};
+    ComplexValue<asymmetric_t> const if_sc_asym{1.0 + 1.0i};
+    ComplexValue<asymmetric_t> const it_sc_asym{2.0 + (2.0i * sqrt(3))};
+
+    CHECK(asym_line.math_model_type() == ComponentType::branch);
+
+    SUBCASE("Voltge error") { CHECK_THROWS_AS(AsymLine(input, 50.0, 10.0e3, 50.0e3), ConflictVoltage); }
+
+    SUBCASE("General") {
+        CHECK(branch.from_node() == 2);
+        CHECK(branch.to_node() == 3);
+        CHECK(branch.from_status() == true);
+        CHECK(branch.to_status() == true);
+        CHECK(branch.branch_status() == true);
+        CHECK(branch.status(BranchSide::from) == branch.from_status());
+        CHECK(branch.status(BranchSide::to) == branch.to_status());
+        CHECK(branch.base_i_from() == doctest::Approx(base_i));
+        CHECK(branch.base_i_to() == doctest::Approx(base_i));
+        CHECK(branch.phase_shift() == 0.0);
+        CHECK(!branch.is_param_mutable());
+    }
+
+    SUBCASE("Symmetric parameters") {
+        // double connected
+        BranchCalcParam<symmetric_t> param = branch.calc_param<symmetric_t>();
+        CHECK(cabs(param.yff() - yff1) < numerical_tolerance);
+        CHECK(cabs(param.ytt() - yff1) < numerical_tolerance);
+        CHECK(cabs(param.ytf() - yft1) < numerical_tolerance);
+        CHECK(cabs(param.yft() - yft1) < numerical_tolerance);
+        // to connected
+        CHECK(branch.update(BranchUpdate{1, false, na_IntS}).topo);
+        param = branch.calc_param<symmetric_t>();
+        CHECK(cabs(param.yff() - 0.0) < numerical_tolerance);
+        CHECK(cabs(param.ytt() - ys1) < numerical_tolerance);
+        CHECK(cabs(param.ytf() - 0.0) < numerical_tolerance);
+        CHECK(cabs(param.yft() - 0.0) < numerical_tolerance);
+        // not connected
+        CHECK(branch.set_status(na_IntS, false));
+        param = branch.calc_param<symmetric_t>();
+        CHECK(cabs(param.yff() - 0.0) < numerical_tolerance);
+        CHECK(cabs(param.ytt() - 0.0) < numerical_tolerance);
+        CHECK(cabs(param.ytf() - 0.0) < numerical_tolerance);
+        CHECK(cabs(param.yft() - 0.0) < numerical_tolerance);
+        // not changing
+        CHECK(!branch.set_status(false, false));
+        // from connected
+        CHECK(branch.set_status(true, na_IntS));
+        param = branch.calc_param<symmetric_t>();
+        CHECK(cabs(param.yff() - ys1) < numerical_tolerance);
+        CHECK(cabs(param.ytt() - 0.0) < numerical_tolerance);
+        CHECK(cabs(param.ytf() - 0.0) < numerical_tolerance);
+        CHECK(cabs(param.yft() - 0.0) < numerical_tolerance);
+    }
+
+    SUBCASE("Asymmetric parameters") {
+        // double connected
+        BranchCalcParam<asymmetric_t> param = asym_line.calc_param<asymmetric_t>();
+        CHECK((cabs(param.yff() - yffa) < numerical_tolerance).all());
```

Comment:
Unrelated to this PR specifically.

@ PGM team: We should create a `check_close`